File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        counter = {}
        
        for c in s:
            if c not in counter:
                counter[c] = 0
            
            counter[c] += 1
            
        
        for i, c in enumerate(s):
            if counter[c] == 1:
                return i
        
        return -1;

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> ch(26,0);
        int n = s.size();
        
        for(int i = 0; i < n; i++) {
            ch[s[i] - 'a']++;
        }
        
        for(int i = 0; i < n; i++) {
            if(ch[s[i] - 'a'] == 1) return i;
        }
        
        return -1;
        
        /*
        unordered_map<char, int> m;
        int n = s.size();
        int ans = INT_MAX;
        
        for(int i = 0; i < n; i++) {
            if(m.find(s[i]) == m.end()) {
                m[s[i]] = i;
            } else {
                m[s[i]] = INT_MAX;
            }
        }
        
        for(auto it : m) {
            ans = min(it.second, ans);
        }
        
        return (ans == INT_MAX) ? -1 : ans;
        */
    }
};